### PR TITLE
GEOMESA-1744 - The geomesa:Query process should support transforms an…

### PR DESCRIPTION
**Pull request by jnh5y**
…d projections

Signed-off-by: Jim Hughes <jnh5y@ccri.com>

### CODE REVIEW

**Comment by jnh5y** [comment]
NB: @aheyne and I might need to test this out some with GeoServer first.  Initially, this can be for comments.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala [hunk 1]
```diff
@@ -38,23 +38,30 @@ class QueryProcess extends LazyLogging {
...
                  min = 0,
                  description = "The filter to apply to the features collection")
-               filter: Filter
-               ): SimpleFeatureCollection = {
+               filter: Filter,
+
+               @DescribeParameter(
+                 name = "properties",
+                 min = 0,
+                 description = "The filter to apply to the features collection")
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala]: description should be updated

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala [hunk 2]
```diff
@@ -78,7 +85,11 @@ class QueryVisitor(features: SimpleFeatureCollection,
   def query(source: SimpleFeatureSource, query: Query) = {
     logger.debug("Running Geomesa query on source type "+source.getClass.getName)
     val combinedFilter = ff.and(query.getFilter, filter)
-    source.getFeatures(combinedFilter)
+    query.setFilter(combinedFilter)
+    if (properties != null) {
+      query.setPropertyNames(properties)
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala]: do we care about any existing transforms in the query?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/query/QueryProcess.scala]: maybe just log a warning if the existing ones aren't null

**Pull request merged**